src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs

```diff
@@ -0,0 +1,377 @@
+        }
+
+        /// <summary>
+        /// The ID of the user that owns the file represented by this entry.
+        /// </summary>
+        /// <remarks>This field is only supported in Unix platforms.</remarks>
+        public int Uid
+        {
+            get => _header._uid;
+            set => _header._uid = value;
+        }
+
+        /// <summary>
+        /// Extracts the current entry to the filesystem.
+        /// </summary>
+        /// <param name="destinationFileName">The path to the destination file.</param>
+        /// <param name="overwrite"><see langword="true"/> if this method should overwrite any existing filesystem object located in the <paramref name="destinationFileName"/> path; <see langword="false"/> to prevent overwriting.</param>
+        /// <exception cref="ArgumentException"><paramref name="destinationFileName"/> is <see langword="null"/> or empty.</exception>
+        /// <exception cref="IOException"><para>The parent directory of <paramref name="destinationFileName"/> does not exist.</para>
+        /// <para>-or-</para>
+        /// <para><paramref name="overwrite"/> is <see langword="false"/> and a file already exists in <paramref name="destinationFileName"/>.</para>
+        /// <para>-or-</para>
+        /// <para>A directory exists with the same name as <paramref name="destinationFileName"/>.</para>
+        /// <para>-or-</para>
+        /// <para>An I/O problem occurred.</para></exception>
+        /// <exception cref="NotSupportedException">Attempted to extract an unsupported entry type.</exception>
+        public void ExtractToFile(string destinationFileName, bool overwrite)
+        {
+            ArgumentException.ThrowIfNullOrEmpty(destinationFileName);
+
+            string? directoryPath = Path.GetDirectoryName(destinationFileName);
+            // If the destination contains a directory segment, need to check that it exists
+            if (!string.IsNullOrEmpty(directoryPath) && !Path.Exists(directoryPath))
+            {
+                throw new IOException(string.Format(SR.IO_PathNotFound_NoPathName, destinationFileName));
+            }
+
+            VerifyOverwriteFileIsPossible(destinationFileName, overwrite);
+
+            switch (EntryType)
+            {
+                case TarEntryType.Directory:
+                case TarEntryType.DirectoryList:
+                    Directory.CreateDirectory(destinationFileName);
+                    break;
+
+                case TarEntryType.RegularFile:
+                case TarEntryType.V7RegularFile:
+                case TarEntryType.ContiguousFile:
+                    // Rely on FileStream's ctor for further checking destinationFileName parameter
+                    FileMode fileMode = overwrite ? FileMode.Create : FileMode.CreateNew;
+
+                    using (FileStream fs = new(destinationFileName, fileMode, FileAccess.Write, FileShare.None))
+                    {
+                        if (DataStream != null)
```

Comment:
Why would anyone want to use this method with `DataStream == null`?

Reply:
The user can freely substitute the `DataStream` of a regular file with any other stream, even a `null` stream. I think a user could say "I want to extract this regular file and create an empty file, but I don't need to create a whole new stream instance for that".